insights/specs/datasources/env.py

```diff
@@ -0,0 +1,95 @@
+"""
+Custom datasources for environments.
+"""
+import re
+import os
+
+from insights.core.context import HostContext
+from insights.core.dr import SkipComponent
+from insights.core.plugins import datasource
+from insights.core.spec_factory import DatasourceProvider
+
+
+@datasource(HostContext)
+def ld_library_path_of_global(broker):
+    """
+    This datasource gets the LD_LIBRARY_PATH enviorment setting for root user,
+    if it's set, then reads the following config files:
+      * /etc/environment
+      * /etc/env.d/*
+      * /etc/profile
+      * /etc/profile.d/*
+      * /etc/bashrc
+      * /etc/bash.bashrc
+      * /root/.bash_profile
+      * /root/.bashrc
+      * /root/.profile
+      * /root/.cshrc
+      * /root/.zshrc
+      * /root/.tcshrc
+
+    When a line likes `export LD_LIBRARY_PATH=$LD_LIBRARY_PATH` exists, the datasource
+    returns the config files that include the config.
+
+    The output of this datasource looks like:
+        /etc/environment
+        /root/.profile
+    Returns:
+        str: Returns a multiline string, each line represent a config file. Not return
+             the value of enviorment, because it might include sensitive information.
+    Raises:
+        SkipComponent: When any exception occurs.
+    """
+    env_name = "LD_LIBRARY_PATH"
+    config_files = [
+        "/etc/environment",
+        "/etc/env.d",
+        "/etc/profile",
+        "/etc/profile.d",
+        "/etc/bashrc",
+        "/etc/bash.bashrc",
+        "/root/.bash_profile",
+        "/root/.bashrc",
+        "/root/.profile",
+        "/root/.cshrc",
+        "/root/.zshrc",
+        "/root/.tcshrc"
+    ]
+    data = []
+
+    for item in config_files:
+        if (os.path.exists(item) and
+                os.path.isfile(item) and
+                _is_env_exported(item, env_name)):
+            data.append(item)
+        elif os.path.exists(item) and os.path.isdir(item):
```

Comment:
Better use `os.walk` to recursively read the files here in case the dir is structured in multiple levels

Reply:
@xiangce Thanks for the review! I intent to use `os.listdir` here. Per the following test:

```
[root@preserve-storage-server1 profile.d]# cat test_1.sh 
export TEST_FILE="/testfile"
[root@preserve-storage-server1 profile.d]# cat test/test.sh 
export TEST_DIR="test_dir"
[root@preserve-storage-server1 profile.d]# env|grep TEST
TEST_FILE=/testfile
```
Looks like only the files under `/etc/profile.d` or `/etc/env.d` directly will be executed.

Reply:
I added a comment here to explain why `listdir` is used here.

Reply:
It's because of the following bash scripts written in the /etc/profile:
For general Linux distribution, it looks like 
```bash
for script in /etc/profile.d/*.sh ; do
        if [ -r $script ] ; then
                . $script
        fi
done
```
For RHEL, it is:
```bash
for i in /etc/profile.d/*.sh /etc/profile.d/sh.local ; do
    if [ -r "$i" ]; then
        if [ "${-#*i}" != "$-" ]; then
            . "$i"
        else
            . "$i" >/dev/null
        fi
    fi
done
```
I'd to say, currently the `listdir` works. Whenever there is some change RHEL officially or customized by customer, things will be changed.

But it doesn't seem that it is worth monitoring such changes for us.